aspnetcore/security/authorization/limitingidentitybyscheme.md

```diff
@@ -14,61 +14,115 @@ uid: security/authorization/limitingidentitybyscheme
 ---
 # Limiting identity by scheme
 
-<a name=security-authorization-limiting-by-scheme></a>
+In some scenarios, such as Single Page Applications, it's common to use multiple authentication methods. For example, your application may use cookie-based authentication to log in and JWT bearer authentication for JavaScript requests. In some cases, you may have multiple instances of an authentication middleware. For example, two cookie middlewares where one contains a basic identity and one is created when a multi-factor authentication has triggered (because the user requested an operation that requires extra security).
 
-In some scenarios, such as Single Page Applications it is possible to end up with multiple authentication methods. For example, your application may use cookie-based authentication to log in and bearer authentication for JavaScript requests. In some cases you may have multiple instances of an authentication middleware. For example, two cookie middlewares where one contains a basic identity and one is created when a multi-factor authentication has triggered because the user requested an operation that requires extra security.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-Authentication schemes are named when authentication middleware is configured during authentication, for example
+An authentication scheme is named when the authentication service is configured during authentication. For example:
 
 ```csharp
-app.UseCookieAuthentication(new CookieAuthenticationOptions()
+public void ConfigureServices(IServiceCollection services)
 {
-    AuthenticationScheme = "Cookie",
-    LoginPath = new PathString("/Account/Unauthorized/"),
-    AccessDeniedPath = new PathString("/Account/Forbidden/"),
-    AutomaticAuthenticate = false
-});
+    // Code omitted for brevity
+
+    services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+        .AddCookie(options => {
+            options.LoginPath = "/Account/Unauthorized/";
+            options.AccessDeniedPath = "/Account/Forbidden/";
+        })
+        .AddJwtBearer();
+```
+
+In the preceding code, two authentication services have been added: one for cookies and one for bearer.
+
+>[!NOTE]
+>When adding multiple authentication middlewares, ensure that no middleware is configured to run automatically. You do this by supplying an argument, such as `CookieAuthenticationDefaults.AuthenticationScheme`, to the `AddAuthentication` method. If you fail to do this, filtering by scheme will not work.
```

Comment:
?? `services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)` makes cookies run automatically. You can override this per endpoint as shown by the Authorize attributes below.